packages/server/modules/core/services/streams/streamAccessService.js

```diff
@@ -13,9 +13,25 @@ const {
   addStreamPermissionsRevokedActivity,
   addStreamInviteAcceptedActivity
 } = require('@/modules/activitystream/services/streamActivity')
+const { getStream } = require('@/modules/core/repositories/streams')
 
 /**
- * Validate that the user has the required permission level (or one above it) for the specified stream
+ * Check if user is a stream collaborator
+ * @param {string} userId
+ * @param {string} streamId
+ * @returns
+ */
+async function isStreamCollaborator(userId, streamId) {
+  const stream = await getStream({ streamId, userId })
```

Comment:
there is a bit of overhead here, looking up the stream is not necessary, could do with just the act table lookup

Reply:
I thought it would be better to re-use a general-purpose existing stream repository function, instead of creating a new one, considering that performance wise they would be more or less the same